common/djangoapps/track/middleware.py

```diff
@@ -204,3 +205,51 @@ def process_response(self, _request, response):
             pass
 
         return response
+
+
+    def truncate_openended_responses(self, event):
+        """
+        Truncates the tracking log entry for problems for which the
+        student response can be long winded.
+
+        event: dict, Constructed from POST and GET request params
+        """
+        event_json = json.loads(event)
+        event_post_keys = event_json['POST'].keys()
+        for key in event_post_keys:
+
+            # problem type with 'student_answer'
+            if key == 'student_answer' and isinstance(event_json['POST'].get('student_answer'), list):
+                truncated_answers = []
+                student_answer = event_json['POST'].get('student_answer')
```

Comment:
Instead of calling `get` twice, let's just memoize this above and reference it from there.

In fact, since we're checking for a specific key here, we shouldn't be doing this inside a loop. Especially given our potential performance concerns, this should be checked independently.

```python
if 'student_answer' in event_json['POST']:
    student_answer = event_json['POST']['student_answer']
    for answer in student_answer:
        chunk_length = TRUNCATION_LENGTH / len(student_answer)
        truncated_answer = answer[:chunk_length]
        truncated_answers.append(truncated_answer) 
    event_json['POST']['student_answer'] = truncated_answers
    return json.dumps(event_json)
for key in event_post_keys:
    if key[:15] == '{"submission":[':
        # ...
        return json.dumps(event_json)
return event[:TRUNCATION_LENGTH]
```